packages/modules/postgresql/src/postgresql-container.test.ts

```diff
@@ -113,3 +113,209 @@ describe("PostgreSqlContainer", { timeout: 180_000 }, () => {
+    expect(result.rows).toHaveLength(1);
+    expect(result.rows[0].name).toEqual("initial data");
+
+    await client.end();
+    await container.stop();
+  });
+
+  it("should handle multiple snapshots", async () => {
+    const container = await new PostgreSqlContainer().start();
+
+    // Connect to the database
+    let client = new Client({
+      connectionString: container.getConnectionUri(),
+    });
+    await client.connect();
+
+    // Create a test table
+    await client.query("CREATE TABLE test_table (id SERIAL PRIMARY KEY, name TEXT)");
+
+    // Close connection before snapshot
+    await client.end();
+
+    // Take first snapshot with empty table
+    await container.snapshot("snapshot1");
+
+    // Reconnect to database
+    client = new Client({
+      connectionString: container.getConnectionUri(),
+    });
+    await client.connect();
+
+    // Add first record
+    await client.query("INSERT INTO test_table (name) VALUES ('data for snapshot 2')");
+
+    // Close connection before snapshot
+    await client.end();
+
+    // Take second snapshot with one record
+    await container.snapshot("snapshot2");
+
+    // Reconnect to database
+    client = new Client({
+      connectionString: container.getConnectionUri(),
+    });
+    await client.connect();
+
+    // Add second record
+    await client.query("INSERT INTO test_table (name) VALUES ('data after snapshots')");
+
+    // Verify we have two records
+    let result = await client.query("SELECT COUNT(*) as count FROM test_table");
+    expect(result.rows[0].count).toEqual("2");
+
+    // Close connection before restore
+    await client.end();
+
+    // Restore to first snapshot (empty table)
+    await container.restore("snapshot1");
+
+    // Reconnect to database
+    client = new Client({
+      connectionString: container.getConnectionUri(),
+    });
+    await client.connect();
+
+    // Verify table is empty
+    result = await client.query("SELECT COUNT(*) as count FROM test_table");
+    expect(result.rows[0].count).toEqual("0");
+
+    // Close connection before restore
+    await client.end();
+
+    // Restore to second snapshot (one record)
+    await container.restore("snapshot2");
+
+    // Reconnect to database
+    client = new Client({
+      connectionString: container.getConnectionUri(),
+    });
+    await client.connect();
+
+    // Verify we have one record
+    result = await client.query("SELECT * FROM test_table");
+    expect(result.rows).toHaveLength(1);
+    expect(result.rows[0].name).toEqual("data for snapshot 2");
+
+    await client.end();
+    await container.stop();
+  });
+
+  it("should throw an error when trying to snapshot postgres system database", async () => {
+    const container = await new PostgreSqlContainer().withDatabase("postgres").start();
+
+    await expect(container.snapshot()).rejects.toThrow(
+      "Cannot restore the postgres system database as it cannot be dropped to be restored"
```

Comment:
Could we change this to `Cannot snapshot` when calling `container.snapshot()`, and `Cannot restore` when calling `container.restore()`?

Reply:
I think I have an idea

Reply:
What do you think of the new error messages? I think they convey the message much better

Reply:
Beautiful, thanks you